test/NameRegistry.t.sol

```diff
@@ -2754,49 +2972,117 @@ contract NameRegistryTest is Test {
+        }
     }
 
-    function testCannotReclaimIfRegistrable(address mod) public {
+    function testCannotReclaimIfRegistrable(address mod, address[4] calldata destinations) public {
         _assumeClean(mod);
+        address[] memory addresses = new address[](5);
+        for (uint256 i = 0; i < destinations.length; i++) {
+            _assumeClean(destinations[i]);
+            addresses[i] = destinations[i];
+        }
+        addresses[4] = mod;
+        _assumeUnique(addresses);
         _grant(MODERATOR_ROLE, mod);
 
+        uint256[] memory tokenIds = new uint256[](4);
+        tokenIds[0] = ALICE_TOKEN_ID;
+        tokenIds[1] = BOB_TOKEN_ID;
+        tokenIds[2] = CAROL_TOKEN_ID;
+        tokenIds[3] = DAN_TOKEN_ID;
+
+        NameRegistry.ReclaimAction[] memory reclaimActions = new NameRegistry.ReclaimAction[](4);
+
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            reclaimActions[i] = NameRegistry.ReclaimAction(tokenIds[i], destinations[i]);
+        }
+
         vm.prank(mod);
         vm.expectRevert(NameRegistry.Registrable.selector);
-        nameRegistry.reclaim(ALICE_TOKEN_ID);
-
-        assertEq(nameRegistry.balanceOf(POOL), 0);
-        vm.expectRevert("ERC721: invalid token ID");
-        assertEq(nameRegistry.ownerOf(ALICE_TOKEN_ID), address(0));
-        assertEq(nameRegistry.expiryOf(ALICE_TOKEN_ID), 0);
-        assertEq(nameRegistry.recoveryOf(ALICE_TOKEN_ID), address(0));
-        assertEq(nameRegistry.recoveryClockOf(ALICE_TOKEN_ID), 0);
+        nameRegistry.reclaim(reclaimActions);
+
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            assertEq(nameRegistry.balanceOf(destinations[i]), 0);
+            vm.expectRevert("ERC721: invalid token ID");
+            assertEq(nameRegistry.ownerOf(tokenIds[i]), address(0));
+            assertEq(nameRegistry.expiryOf(tokenIds[i]), 0);
+            assertEq(nameRegistry.recoveryOf(tokenIds[i]), address(0));
+            assertEq(nameRegistry.recoveryClockOf(tokenIds[i]), 0);
+        }
     }
 
-    function testCannotReclaimUnlessModerator(address alice, address notModerator, address recovery) public {
-        _assumeClean(alice);
+    function testCannotReclaimUnlessModerator(
+        address[4] calldata users,
+        address[4] calldata destinations,
+        address notModerator,
+        address[4] calldata recoveryAddresses
+    ) public {
         _assumeClean(notModerator);
-        _assumeClean(recovery);
+        address[] memory addresses = new address[](13);
+        for (uint256 i = 0; i < users.length; i++) {
+            _assumeClean(users[i]);
+            _assumeClean(destinations[i]);
+            _assumeClean(recoveryAddresses[i]);
+            addresses[i] = users[i];
+            addresses[i + 4] = destinations[i];
+            addresses[i + 8] = recoveryAddresses[i];
+        }
+        addresses[12] = notModerator;
+        _assumeUnique(addresses);
+
+        bytes16[] memory fnames = new bytes16[](4);
```

Comment:
could we declare the fnames and tokenIds array once in the file and use them in every method? re-adding them everywhere is somewhat verbose